polaris-react/src/components/SettingToggle/SettingToggle.tsx

```diff
@@ -1,39 +1,150 @@
 import React, {useMemo} from 'react';
 
 import type {ComplexAction} from '../../types';
-import {SettingAction} from '../SettingAction';
 import {buttonFrom} from '../Button';
-import {LegacyCard} from '../LegacyCard';
 import {globalIdGeneratorFactory} from '../../utilities/unique-id';
+import {Text} from '../Text';
+import {Badge} from '../Badge';
+import {useMediaQuery} from '../../utilities/media-query';
+import {AlphaCard} from '../AlphaCard';
+import {AlphaStack} from '../AlphaStack';
+import {Inline} from '../Inline';
+import {Box} from '../Box';
+import {useI18n} from '../../utilities/i18n';
+import {WithinContentContext} from '../../utilities/within-content-context';
 
 export interface SettingToggleProps {
-  /** Inner content of the card */
-  children?: React.ReactNode;
-  /** Card header actions */
+  /** The action that toggles the setting on or off */
   action?: ComplexAction;
-  /** Sets toggle state to activated or deactivated */
+  /** Additional content informing merchants of the impact or requirements of the setting */
+  children?: React.ReactNode;
+  /** A description of what the setting controls */
+  description?: string;
+  /** Whether or not the setting is enabled */
   enabled?: boolean;
+  /** The help link rendered in the setting */
+  helpLink?: React.ReactNode;
+  /** The name of the setting */
+  title?: string;
```

Comment:
Hey @oluwatimio. I am not aligned with these changes. You are adding a lot of props and translations to the component. Polaris should focus on creating small composable patterns. Additional props is more maintenance, design opinions, documentation and examples.

I think this component would be better outside of Polaris. We have recently been figuring out patterns and I think this is a good example of what a pattern could be.